trestle/core/catalog_interface.py

```diff
@@ -650,85 +651,200 @@ def write_catalog_as_markdown(self, context: ControlContext, part_id_map: Dict[s
-                        if key not in control_param_dict:
-                            pop_list.append(key)
-                    for pop in pop_list:
-                        new_context.yaml_header[const.SET_PARAMS_TAG].pop(pop)
-            _, group_title, _ = self.get_group_info_by_control(control.id)
-            # control could be in sub-group of group so build path to it
-            group_dir = new_context.md_root
-            control_path = self.get_control_path(control.id)
-            for sub_dir in control_path:
-                group_dir = group_dir / sub_dir
-                if not group_dir.exists():
-                    group_dir.mkdir(parents=True, exist_ok=True)
+                for set_param in as_list(new_context.control_implementation.set_parameters):
+                    _update_values(set_param)
+                for imp_req in as_list(new_context.control_implementation.implemented_requirements):
+                    if imp_req.control_id == control.id:
+                        for set_param in as_list(imp_req.set_parameters):
+                            _update_values(set_param)
+
+                if control_param_dict:
+                    new_context.yaml_header[const.PARAM_VALUES_TAG] = {}
+                    for key, param in control_param_dict.items():
+                        new_context.yaml_header[const.PARAM_VALUES_TAG][key] = none_if_empty(
+                            ControlInterface.param_to_str(param, ParameterRep.VALUE_OR_EMPTY_STRING)
+                        )
+
+                self._write_control_into_dir(new_context, control, part_id_map, [])
+
+    def write_catalog_as_catalog(self, context: ControlContext, part_id_map: Dict[str, Dict[str, str]]) -> None:
+        """Write the catalog as a simple catalog."""
+        required_section_list = context.required_sections.split(',') if context.required_sections else []
+        allowed_section_list = context.allowed_sections.split(',') if context.allowed_sections else []
+
+        # write out the controls
+        for control in self.get_all_controls_from_catalog(True):
+            # here we do special handling of how set-parameters merge with the yaml header
+            new_context = ControlContext.clone(context)
+
+            control_param_dict = ControlInterface.get_control_param_dict(control, False)
+            set_param_dict: Dict[str, str] = {}
+            for param_id, param_dict in control_param_dict.items():
+                # if the param is in the full_param_dict, load its contents first and mark as profile-values
+                display_name = ''
+                # if the profile doesnt change this param at all, show it in the header with values
+                tmp_dict = ModelUtils.parameter_to_dict(param_dict, True)
+                values = tmp_dict.get('values', None)
+                new_dict = {'id': param_id, 'values': values}
+                new_dict.pop('id')
```

Comment:
Why adding id and then popping right the way?

Reply:
fixed - it was meant to show it was a dict like the others initially, with an id.